### PR TITLE
[OFFAPPS-1020] Move branches and update app_migrator to use built-in scaffold

### PR DESCRIPTION
**Pull request by yu-tian113**
:v:

/cc @zendesk/vegemite 

### Tasks
- [x] Move app_scaffold [master branch](https://github.com/zendesk/app_scaffold) into [app_migrator repository](https://github.com/zendesk/zendesk_app_migrator)
- [x] Update [app_migrator](https://github.com/zendesk/zendesk_app_migrator) to use built-in migration scaffold

### References
* JIRA: https://zendesk.atlassian.net/browse/OFFAPPS-1020

### Risks
* [low] Make sure app migrator is working and using built-in scaffold

### CODE REVIEW

File: lib/app_scaffold/dist/assets/iframe.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  </head>
+
+  <body>
+    <section data-main class="main">
+      <img class="loader" src="dot.gif"/>
+    </section>
+
+    <script type="text/javascript" src="https://assets.zendesk.com/apps/sdk/2.0/zaf_sdk.js"></script>
+    <script type="text/javascript" src="app.js"></script>
+</body>
```
Comment (keitheous) [inline lib/app_scaffold/dist/assets/iframe.html]: 👉

Comment (vaevictis) [reply lib/app_scaffold/dist/assets/iframe.html]: ! I was like,
> What does this mean?

It's about indentation right?

Comment (keitheous) [reply lib/app_scaffold/dist/assets/iframe.html]: Oh yes sorry. The indentation doesn't look aligned here.

File: src/app_scaffold/lib/javascripts/base_app.js [hunk 2]
```diff
@@ -56,6 +56,14 @@ function registerHelpers(app) {
   });
 }
 
+function xmlParserDecorator(fn) {
+  return function(...args) {
+    const DOMPARSER = new DOMParser()
+    args[0] = DOMPARSER.parseFromString(args[0], 'application/xml')
+    fn(...args)
```
Comment (ocke) [inline src/app_scaffold/lib/javascripts/base_app.js]: Compare with jQuery xml parser

File: src/app_scaffold/lib/javascripts/view.js [hunk 3]
```diff
@@ -4,6 +4,12 @@ import _ from 'lodash';
 class View {
   constructor(opts = {}) {
     this.afterRender = opts.afterRender;
+    this._renderLayout()
+  }
+
+  _renderLayout() {
+    let layout = require(`../../src/templates/layout.hdbs`);
+    layout && $('[data-main]').replaceWith(layout)
```
Comment (ocke) [inline src/app_scaffold/lib/javascripts/view.js]: remove <header> in the layout.hbs

File: src/app_scaffold/.gitignore [hunk 4]
```diff
@@ -0,0 +1,9 @@
+node_modules
+bower_components
+dist/assets/index.html
```
[the remark this answers is not in the document]
Comment (yu-tian113) [reply src/app_scaffold/.gitignore]: `dist/` is deliberately committed for storing manifest.json and some other assets, they are not moved into `dist` from `src` as we do normally.
Consider we only *moving* this migration scaffold from another repo, I reckon just leave it as it is.

File: src/app_scaffold/lib/javascripts/base64.js [hunk 5]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
```
[the remark this answers is not in the document]
Comment (yu-tian113) [reply src/app_scaffold/lib/javascripts/base64.js]: Same as above, this is used for encode http headers, I reckon it's optional helper

File: README.md [hunk 6]
```diff
@@ -25,11 +25,8 @@ The migrator executes a series of tasks to:
 
 #### App Scaffold
 
-The migrator has a hard dependency on the public
-[App Scaffold](https://github.com/zendesk/app_scaffold) project.
-
-When a v1 app is migrated, we leverage the features already built into the App
-Scaffold to transpile v1 app assets for v2.
+Please refer to [App Migration Scaffold](https://github.com/zendesk/zendesk_app_migrator/tree/master/lib/app_scaffold/README.md)
+for detailed documentation on developing, testing, building and deploying applications migrated to V2.
```
[the remark this answers is not in the document]
Comment (yu-tian113) [reply README.md]: 👍

**Comment by yu-tian113** [comment]
Hey @danielbreves @ocke @hoangtrvu 
As discussed in the PR session, I revert the new feature updates.
This PR only includes adding local migration scaffold and update path to local scaffold
Thanks!

File: src/app_scaffold/package.json [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    "karma-chrome-launcher": "^2.2.0",
+    "karma-jasmine": "^1.1.0",
+    "karma-phantomjs-launcher": "^1.0.4",
+    "karma-webpack": "^2.0.4",
+    "loader-utils": "^1.1.0",
+    "node-sass": "^4.5.3",
+    "sass-loader": "^6.0.6",
+    "style-loader": "^0.18.2",
+    "url-loader": "^0.5.9",
+    "webpack": "^3.5.1"
```
Comment (keitheous) [inline src/app_scaffold/package.json]: do we need webpack-cli?

File: src/app_scaffold/lib/javascripts/base64.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (danielbreves) [reply src/app_scaffold/lib/javascripts/base64.js]: I think at some point we had a helper for IE < 10, so we kept it here to make it easier for v1 apps who were relying on it.

File: src/app_scaffold/package.json (hunk 7 above)
Comment (yu-tian113) [reply src/app_scaffold/package.json]: Yes, in package.json
```
"scripts": {
    "build": "webpack -p",
    "watch": "webpack --watch"
},
```

Comment (yu-tian113) [reply src/app_scaffold/package.json]: @ocke, 👍, good to know

**Pull request merged**